src/main/java/seedu/address/logic/parser/ShowFriendCommandParser.java

```diff
@@ -21,17 +24,38 @@
      * @throws ParseException If the input entered is invalid.
      */
     public ShowFriendCommand parse(String args) throws ParseException {
-        ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME) || !argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ShowFriendCommand.MESSAGE_USAGE));
+        Pattern p = Pattern.compile("^-?\\d*\\.{0,1}\\d+$"); //Regex to match all numeric Strings
+        boolean isShowByIndex = isNumeric(p, args);
+
+        if (isShowByIndex) {
+            try {
+                Index index = ParserUtil.parseIndex(args);
+                return new ShowFriendCommand(index);
+            } catch (ParseException pe) {
+                throw new ParseException(MESSAGE_INDEX_IS_NOT_NON_ZERO_UNSIGNED_INTEGER);
```

Comment:
Any reason you're catching and throwing the same thing?

Reply:
Fixed!